src/configfetcher.cpp

```diff
@@ -71,6 +71,32 @@ ConfigFetcher::ConfigFetcher(const string& sdkKey, shared_ptr<ConfigCatLogger> l
                 return !closed;
             }
     });
+
+    if(options.sslOptions)
+    {
+        // TODO: due to a bug in CPR requirest the cpr::SslOptions object to be kept alive. to discuss with Peter
+        static cpr::SslOptions sslOpts;
```

Comment:
The `cpr::Session::SetSslOptions` [implementation](https://github.com/libcpr/cpr/blob/e7618a6afedd42e995fb728bba385f8af84ca7cc/cpr/session.cpp#L478C1-L478C1) sets the data from `ca_buffer` to `CURLOPT_SSL_CTX_DATA`. However the `std::string` gets destroyed before the function to evaluate the certificate is called.

To make this work we have to either:
- keep this instance of `cpr::SslOptions` alive: by making it a `std::shared_ptr` and keeping the pointer.
OR:
- delay the setting of `cpr::Session::SslOptions` until the `configcat::ConfigFetcher::fetch` to make they are evaluated before destroyed: this would require to access the `configcat::ConfigCatOptions` right before fetching.

Reply:
This was also removed since we no longer use CPR